msupply-foundation/open-msupply
# 2929 add reason step

Fixes #2929

# 👩🏻‍💻 What does this PR do? 
 

Adds the second step of the return items modal, and mocks the insert mutation on Ok:
![Screenshot 2024-02-14 at 2 04 31 PM](https://github.com/msupply-foundation/open-msupply/assets/55115239/6ba91d79-a76a-4a5c-a9e8-6535aaa3f545)


# 🧪 How has/should this change been tested? 



- Open an existing/create a new Inbound Shipment
- Select some of the shipment lines
- Select the 'Return to supplier' action
- The Return Items modal opens (populated with a dummy line)
- Click Ok & Next
- See/edit the Reason and Comment inputs for the return lines
- Clicking OK calls the mutation and closes the modal


## 💌 Any notes for the reviewer?



There aren't many frontend tests beyond the common package... is this intentional? Otherwise feeling I should be wrapping tests around the hooks/rendering logic 😅
